### ISSUE TITLE
Use AWS metadata IP in CloudInstanceConfig 

### ISSUE THREAD

**Issue by sfussenegger** [Netflix/eureka#715]
I'm trying to get Eureka running in AWS EC2 Container Service (ECS).

Unfortunately, `CloudInstanceConfig` relies on `InetAddress.getLocalHost().getHostAddress()` to get the IP address. In case of Docker, this is the IP inside the Docker network (typically something like `172.17.0.2`) which isn't very helpful.

Using `MetaDataKey.localIpv4` should fix this problem without changing current behavior:

```java
@Override
public String getIpAddress() {
    return ((AmazonInfo) info).get(MetaDataKey.localIpv4);
}
```

here's the current implementation of `getHostName(..)`:

```java
@Override
public String getHostName(boolean refresh) {
    if (refresh) {
        refreshAmazonInfo();
    }
    return ((AmazonInfo) info).get(MetaDataKey.publicHostname);
}
```

Alternatively, there could be a new property to enable this behavior.

**Comment by tbak** [comment]
CloudInstanceConfig depends on Amazon metadata. If access to metadata
fails, and flag 'eureka.validateInstanceId' is set to false, as a fallback
a local host address is read. Possibly this is happening in your case.
You can always override the default CloudInstanceConfig implementation, and
provide your custom version that better handles docker environment.
/Tomasz

On Fri, Dec 4, 2015 at 7:16 AM, Stefan Fussenegger <notifications@github.com

**Comment by qiangdavidliu** [comment]
There is actually some work in progress that will address this exact issue
(for a different use case). The changes should make it into the next
release.

**Comment by jhmartin** [comment]
Any sort of ETA on the release? I am interested in running clients in ECS as well.

**Comment by qiangdavidliu** [comment]
Sorry for not updating. This is actually already fixed and released in release 1.3.7. Will update the release notes also.
